metagpt/utils/common.py

```diff
@@ -305,3 +305,9 @@ def parse_recipient(text):
     pattern = r"## Send To:\s*([A-Za-z]+)\s*?"  # hard code for now
     recipient = re.search(pattern, text)
     return recipient.group(1) if recipient else ""
+
+
+def ensure_trailing_slash(url):
+    if not url:
+        return url
+    return url if url.endswith("/") else url + "/"
```

Comment:
The code has been removed because v1 provides a [function](https://github.com/openai/openai-python/blob/e36956673d9049713c91bca6ce7aebe58638f483/src/openai/_base_client.py#L358) with the same functionality: